release-notes/cabal-install-3.16.0.0.md

```diff
@@ -0,0 +1,170 @@
+
+  Now:
+
+  * If you specify `--enable-shared` then shared objects are built.
+  * If you specify `--disabled-shared` then shared objects are not built.
+  * If you don't explicitly specify whether you want to build shared libraries then
+    * `--enable-executable-dynamic` will automatically turn on building shared libraries
+    * `--enable-executable-dynamic --enable-profiling` will automatically turn on building
+      shared profiling libraries (if supported by your compiler).
+
+- Working directory support for Cabal [#9702](https://github.com/haskell/cabal/issues/9702) [#9718](https://github.com/haskell/cabal/pull/9718)
+
+  The Cabal library is now able to handle a passed-in working directory, instead
+  of always relying on the current working directory of the parent process.
+
+  In order to achieve this, the `SymbolicPath` abstraction was fleshed out, and
+  all fields of `PackageDescription` that, if relative, should be interpreted
+  with respect to e.g. the package root, use `SymbolicPath` instead of `FilePath`.
+
+  This means that many library functions in `Cabal` take an extra argument of type
+  `Maybe (SymbolicPath CWD (Dir "Package))`, which is an optional (relative or
+  absolute) path to the package root (if relative, relative to the current working
+  directory). In addition, many functions that used to manipulate `FilePath`s now
+  manipulate `SymbolicPath`s, require explicit conversion using e.g. `getSymbolicPath`.
+
+  To illustrate with file searching, the `Cabal` library defines:
+
+  ```haskell
+  findFileCwd
+    :: forall dir1 dir2 file
+     . Verbosity
+    -> Maybe (SymbolicPath CWD (Dir dir1))
+
+    -> [SymbolicPath dir1 (Dir dir2)]
+
+    -> RelativePath dir2 File
+
+    -> IO (SymbolicPath dir1 File)
+  ```
+
+  See Note [Symbolic paths] in `Distribution.Utils.Path` for further information
+  on the design of this API.
+
+- `curl` transport now supports Basic authentication [#10089](https://github.com/haskell/cabal/pull/10089)
+
+  - The `curl` HTTP transport previously only supported the HTTP Digest
+    authentication scheme.  Basic authentication is now supported
+    when using HTTPS; Curl will use the scheme offered by the server.
+    The `wget` transport already supports HTTPS.
+
+- Enhance error detection for cabal root project files, including broken symlinks [#9937](https://github.com/haskell/cabal/issues/9937) [#10103](https://github.com/haskell/cabal/pull/10103)
+
+  - Added proper detection and reporting for issues with cabal root project files. Previously, these files were silently ignored if they were broken symlinks. Now, `cabal` will exit
+  with an error in such case.
+
+- Let cabal init remember chosen language within current session [#10096](https://github.com/haskell/cabal/issues/10096) [#10115](https://github.com/haskell/cabal/pull/10115)
+
+  When cabal init asks for a language, the last choice will be used as the new default for the current session.
```

Comment:
```suggestion
  When `cabal init` asks for a language, the last choice made will be used as the new default for the current prompt.
```

Reply:
To be honest, I still don't understand what a "prompt" or a "session" mean here...

Reply:
I understand "session" to be "a particular run of `cabal init`" and "prompt" to mean the current question it is asking (and am surprised you didn't understand the latter; it's had that meaning since at least the 1980s).